autogluon/task/text_prediction/column_property.py

```diff
@@ -0,0 +1,636 @@
+import abc
+import pandas as pd
+import numpy as np
+import json
+import collections
+from typing import List, Optional, Union, Tuple, Hashable
+from . import constants as _C
+from autogluon_contrib_nlp.base import INT_TYPES, BOOL_TYPES
+from autogluon_contrib_nlp.data.vocab import Vocab
+
+__all__ = ['CategoricalColumnProperty',
+           'TextColumnProperty',
+           'NumericalColumnProperty',
+           'EntityColumnProperty']
+
+
+class ColumnProperty(abc.ABC):
```

Comment:
We should discuss more about this functionality and see if there is overlap with FeatureMetadata and Feature Generators in #579.

Reply:
Let me revise it later. Currently, these APIs are not exposed.